genson/generator.py

```diff
@@ -193,28 +197,26 @@ def _add_properties(self, properties, func):
             getattr(self._properties[prop], func)(val)
 
     def _add_items(self, items, func):
-        if self.merge_arrays:
+        if self._kwargs['merge_arrays']:
             self._add_items_merge(items, func)
         else:
             self._add_items_sep(items, func)
 
     def _add_items_merge(self, items, func):
-        if items:
-            if not self._items:
-                self._items.append(Schema())
-
-            method = getattr(self._items[0], func)
+        if not self._items:
+            self._items = Schema(**self._kwargs)
+        method = getattr(self._items, func)
+        if isinstance(items, list):
             for item in items:
                 method(item)
+        else:
+            method(items)
 
     def _add_items_sep(self, items, func):
         for item in items:
-            subschema = Schema()
+            subschema = Schema(**self._kwargs)
             getattr(subschema, func)(item)
-
-            # only add schema if it's not already there.
-            if subschema not in self._items:
-                self._items.append(subschema)
+            self._items.append(subschema)
```

Comment:
This doesn't actually merge two objects' schemas in a way that's compatible with both. Only the first one will be fully served.